### PR TITLE
quic: attempting to fix non-linux QUIC builds

### PR DESCRIPTION
**Pull request by alyssawilk**
Upstream HTTP/3 android build is breaking becasue the udp gso libraries are bazel-selected out but the includes aren't:
https://github.com/envoyproxy/envoy-mobile/pull/1717/checks?check_run_id=3487450075

I _think_ this will fix (or at least get us to the next compile error)

Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: hopefully fixing android build

### CODE REVIEW

**Comment by alyssawilk** [comment]
api_compat failure is a known failure (which I can merge past)

File: source/common/quic/udp_gso_batch_writer.h [hunk 1]
```diff
@@ -18,8 +18,11 @@
 #pragma GCC diagnostic ignored "-Wsuggest-override"
 #endif
 
+#if defined(__linux__)
```
Comment (danzh2010) [inline source/common/quic/udp_gso_batch_writer.h:21]: Does this fix the build? There are other places in this file refers to quic classes.
Why not excluding this header file completely in the BUILD if the platform is not Linux?

Comment (alyssawilk) [reply source/common/quic/udp_gso_batch_writer.h:21]: right, sorry, I had commented out most of the file in a prior build which got lost.

So I can add that back, moving the endif to the bottom of the file, or we could have one file for setting UDP_GSO_BATCH_WRITER_COMPILETIME_SUPPORT  and a second file for use of it.

which would you prefer?

Comment (danzh2010) [reply source/common/quic/udp_gso_batch_writer.h:21]: I'm okay either way, maybe the latter is easier to understand?

File: source/server/listener_impl.cc [hunk 2]
```diff
@@ -32,6 +32,10 @@
 
 #ifdef ENVOY_ENABLE_QUIC
 #include "source/common/quic/active_quic_listener.h"
+#include "source/common/quic/gso_defines.h"
+#endif // ENVOY_ENABLE_QUIC
+
+#ifdef UDP_GSO_BATCH_WRITER_COMPILETIME_SUPPORT
```
Comment (danzh2010) [inline source/server/listener_impl.cc:38]: If you #ifdef UDP_GSO_BATCH_WRITER_COMPILETIME_SUPPORT in udp_gso_batch_writer.h, this ifdef is not needed.

Comment (alyssawilk) [reply source/server/listener_impl.cc:38]: well right now the file is compile-selected out in the build.
we could leave it in the build and #ifdef out the whole contents, but this seems cleaner to me?

Comment (alyssawilk) [reply source/server/listener_impl.cc:37]: also I'm _super_ baffled by windows breakage

source/server/listener_impl.cc(39): fatal error C1083: Cannot open include file: 'source/common/quic/udp_gso_batch_writer.h': No such file or directory


but it should only see that if UDP_GSO_BATCH_WRITER_COMPILETIME_SUPPORT is 1 which should only happen on linux? What am missing? :-/

Comment (alyssawilk) [reply source/server/listener_impl.cc:37]: oops nm, PEBKAC, fixed.

Comment (danzh2010) [reply source/server/listener_impl.cc]: Agreed this is cleaner. I'm fine either way.
BTW, udp_gso_batch_writer.h is already ifdef'ing out the whole contents, isn't it?

File: source/server/listener_impl.cc [hunk 3]
```diff
@@ -32,9 +32,14 @@
...
 #ifdef ENVOY_ENABLE_QUIC
 #include "source/common/quic/active_quic_listener.h"
+#include "source/common/quic/gso_defines.h"
+#endif // ENVOY_ENABLE_QUIC
+
+#if UDP_GSO_BATCH_WRITER_COMPILETIME_SUPPORT
 #include "source/common/quic/udp_gso_batch_writer.h"
 #endif
 
+
```
Comment (danzh2010) [inline source/server/listener_impl.cc:42]: extra line?

**Review by danzh2010 — APPROVED** [review]

File: source/server/listener_impl.cc (hunk 2 above)
Comment (alyssawilk) [reply source/server/listener_impl.cc]: uggggh, you're totally right, which means I've been attacking the wrong problem from the start.
Thanks for saving me the commit + mobile update + realizing this didn't solve my problem 
trying a new workaround per your offline suggestions!

**Review by junr03 — APPROVED** [review]

**Pull request merged**